crates/polars-core/src/series/arithmetic/borrowed.rs

```diff
@@ -171,6 +175,168 @@ impl NumOpsDispatchInner for FixedSizeListType {
+    let left_as_list = left.as_any().downcast_ref::<LargeListArray>();
+    let right_as_list = right.as_any().downcast_ref::<LargeListArray>();
+    match (left_as_list, right_as_list) {
+        (Some(left), Some(right)) => {
+            left.offsets() == right.offsets() && lists_same_shapes(left.values(), right.values())
+        },
+        (None, None) => left.len() == right.len(),
+        _ => false,
+    }
+}
+
+impl ListChunked {
+    fn arithm_helper(
+        &self,
+        rhs: &Series,
+        op: &dyn Fn(&Series, &Series) -> PolarsResult<Series>,
+    ) -> PolarsResult<Series> {
+        polars_ensure!(
+            self.len() == rhs.len(),
+            InvalidOperation: "can only do arithmetic operations on Series of the same size; got {} and {}",
+            self.len(),
+            rhs.len()
+        );
+
+        thread_local! {
+            static HAS_NULLS: Cell<bool> = const { Cell::new(false) };
+        };
+
+        let orig_has_nulls = HAS_NULLS.get();
+
+        let mut has_nulls = orig_has_nulls;
+        if !has_nulls {
+            for chunk in self.chunks().iter() {
+                if maybe_list_has_nulls(chunk) {
+                    has_nulls = true;
+                    break;
+                }
+            }
+        }
+        if !has_nulls {
+            for chunk in rhs.chunks().iter() {
+                if maybe_list_has_nulls(chunk) {
+                    has_nulls = true;
+                    break;
+                }
+            }
+        }
+        if has_nulls {
+            // A slower implementation since we can't just add the underlying
+            // values Arrow arrays. Given nulls, the two values arrays might not
+            // line up the way we expect.
+
+            // This can be recursive, so preserve the knowledge that there
+            // were nulls. Unfortunately get_leaf_array() and explode()
+            // don't work on the Series that come out of amortized_iter(),
+            // so we need to stick to this code path.
+            HAS_NULLS.set(true);
+            scopeguard::defer! { HAS_NULLS.set(orig_has_nulls); };
+
+            let mut result = self.clear();
+            let combined = self.amortized_iter().zip(rhs.list()?.amortized_iter()).map(|(a, b)| {
+                let (Some(a_owner), Some(b_owner)) = (a, b) else {
+                    // Operations with nulls always result in nulls:
+                    return Ok(Series::full_null(self.name().clone(), 1, self.dtype()));
+                };
+                let a = a_owner.as_ref();
+                let b = b_owner.as_ref();
+                polars_ensure!(
+                    a.len() == b.len(),
+                    InvalidOperation: "can only do arithmetic operations on lists of the same size; got {} and {}",
+                    a.len(),
+                    b.len()
+                );
+                op(a, b).and_then(|s| s.implode()).map(Series::from)
```

Comment:
I think we can keep `ListChunked` here. Instead of going int `Series` and then unpacking `List`  later.

Reply:
OK will try to fix that.